x-pack/solutions/observability/plugins/infra/public/components/asset_details/hooks/use_asset_details_render_props.ts

```diff
@@ -6,20 +6,34 @@
  */
 
 import createContainer from 'constate';
+import { useEffect, useMemo } from 'react';
 import type { AssetDetailsProps } from '../types';
 import { useAssetDetailsUrlState } from './use_asset_details_url_state';
 import { useMetadataStateContext } from './use_metadata_state';
+import { useTimeRangeMetadataContext } from '../../../hooks/use_time_range_metadata';
 
 export type UseAssetDetailsRenderProps = Pick<
   AssetDetailsProps,
   'entityId' | 'entityName' | 'entityType' | 'overrides' | 'renderMode'
 >;
 
 export function useAssetDetailsRenderProps(props: UseAssetDetailsRenderProps) {
-  const [urlState] = useAssetDetailsUrlState();
+  const [urlState, setUrlState] = useAssetDetailsUrlState();
   const { metadata } = useMetadataStateContext();
+  const { data: timeRangeMetadata } = useTimeRangeMetadataContext();
   const { entityId, entityName, entityType, ...rest } = props;
 
+  const schema = useMemo(() => {
+    if (!timeRangeMetadata) return null;
+    return timeRangeMetadata.schemas[0];
+  }, [timeRangeMetadata]);
+
+  useEffect(() => {
```

Comment:
I'm unsure if this is the best place to set the schema in the url state
Any thoughts @crespocarlos ?